arangod/Metrics/MetricsFeature.cpp

```diff
@@ -677,10 +652,73 @@ void MetricsFeature::toPrometheus(std::string& result, bool v2) const {
   if (engineName == RocksDBEngine::EngineName) {
     es.getStatistics(result, v2);
   }
+  if (cm.isEnabled()) {
+    cm.toPrometheus(result, _globalLabelsStr);
+  }
 }
 
 ServerStatistics& MetricsFeature::serverStatistics() noexcept {
   return *_serverStatistics;
 }
 
+constexpr auto kCoordinatorMetrics =
+    frozen::make_unordered_set<frozen::string>({
+        "arangosearch_link_stats",
+        "arangosearch_num_failed_commits",
+        "arangosearch_num_failed_cleanups",
+        "arangosearch_num_failed_consolidations",
+        "arangosearch_commit_time",
+        "arangosearch_cleanup_time",
+        "arangosearch_consolidation_time",
+    });
+
+void MetricsFeature::toVPack(VPackBuilder& builder, bool coordinator) const {
+  builder.openArray(true);
+  {
+    std::lock_guard guard(_lock);
+    unsafeInitTo();
+    for (auto const& i : _registry) {
+      auto const name = i.second->name();
+      if (kSuppressionsV2.count(name) ||
+          (coordinator && !kCoordinatorMetrics.count(name))) {
+        continue;
+      }
+      i.second->toVPack(server(), builder);
+    }
+  }
+  builder.close();
+}
+
+void MetricsFeature::unsafeInitTo() const {
```

Comment:
This function seems to be some sort of an init function, which will initialize some internal state only once after server startup. Maybe we can call this once in `MetricsFeature::start()` or so, to avoid having to use the exclusive `_lock` every time we are serializing the metrics.

Reply:
Maybe, I just move some code to function